### PR TITLE
Feature/update data flow

### PR DESCRIPTION
**Pull request by MizukiSonoko**
## What is this pull request?
Apply PeerCommunicationService to new data follow  

- PeerCommunication Service has  on_proposal() and on_commit() only
- PeerCommunication Service requires ordering_service and consensus_service only
- TransactionProcessor has only transaction_handle()
- TransactionProcessor requires stateless validator and crypto provider only
- stub_query_processor -> query_processor_stub

## Why do you implement it? Why do we need this pull request?
In now, little old interfaces

### CODE REVIEW

File: irohad/torii/processor/impl/transaction_processor_stub.cpp [hunk 1]
```diff
@@ -21,14 +21,13 @@ namespace iroha {
...
-    using network::PeerCommunicationService;
     using model::TransactionResponse;
     using model::ModelCryptoProvider;
 
     TransactionProcessorStub::TransactionProcessorStub(
-        const StatelessValidator &validator, PeerCommunicationService &service,
+        const StatelessValidator &validator,
         ModelCryptoProvider &provider)
-        : validator_(validator), service_(service), provider_(provider) {
+        : validator_(validator), provider_(provider) {
```
Comment (takemiyamakoto) [inline irohad/torii/processor/impl/transaction_processor_stub.cpp]: provider is not very clear. cryptoProvider is better.

File: irohad/main/iroha-main.cpp [hunk 2]
```diff
@@ -61,16 +61,14 @@ int main(int argc, char *argv[]) {
...
+    consensus_service);
   iroha::torii::TransactionProcessorStub tp(stateless_validator,
-                                            peer_communication_service,
                                             crypto_provider);
   iroha::torii::QueryProcessorStub qp(ametsuchi, ametsuchi);
 
   iroha::torii::ToriiStub torii(tp, qp);
-  // shows required order of execution, since callbacks are called synchronously
-  peer_communication_service.subscribe_on_proposal();
+  // shows required order of execution, since callbacks are called synchronousl
```
Comment (takemiyamakoto) [inline irohad/main/iroha-main.cpp]: synchronousl→synchronously

File: irohad/network/network_api.h [hunk 3]
```diff
@@ -45,23 +45,10 @@ namespace iroha {
...
-      virtual void propagate_transaction(const model::Transaction &tx) = 0;
-    };
-
-    /**
      * Public API interface for communication between current peer and other
      * peers in a network
      */
-    class PeerCommunicationService : public TransactionPropagator,
-                                     public ConsensusListener {
+    class PeerCommunicationService : public ConsensusListener {
```
Comment (satellitex) [inline irohad/network/network_api.h]: What is this?
I think we don't use PeerCommunicationService.

File: irohad/network/peer_communication_stub.hpp [hunk 4]
```diff
@@ -19,41 +19,30 @@
...
     class PeerCommunicationServiceStub : public PeerCommunicationService {
      public:
       PeerCommunicationServiceStub(
-          ametsuchi::Ametsuchi &storage,
-          validation::StatefulValidator &stateful_validator,
-          validation::ChainValidator &chain_validator,
-          ordering::OrderingService &orderer,
-          consensus::ConsensusService &consensus,
-          model::ModelCryptoProvider &crypto_provider);
+          ordering::OrderingServiceStub &orderer,
```
Comment (satellitex) [inline irohad/network/peer_communication_stub.hpp]: Why do you modify here?

**Pull request merged**